compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -1043,6 +1043,20 @@ extern "C" void LLVMRustPrintPasses() {
   PB.printPassNames(outs());
 }
 
+// from https://github.com/llvm/llvm-project/blob/7021182d6b43de9488ab70de626192ce70b3a4a6/llvm/lib/Object/IRSymtab.cpp#L48-L57
+static const char *PreservedSymbols[] = {
```

Comment:
Duplicating this piece of state from IRSymtab.cpp makes me nervous. How is anyone going to know that they need to watch that file and update this array if changes are made over there?

Is there no better way to architect this to avoid that duplication?

Its possible that we have no other option here; I skimmed IRSymtab.cpp and didn't see any immediate way to access that hidden state. We could contribute a patch upstream to allow such observation, or even rearchitect their work to move the two strings to a separate .def file analogous to RuntimeLibcalls.def...

but I figured I wanted to ask the question first.

Reply:
First, there will be no more updates in LLVM 17 and lower.

If `InternalizePass` should consider built-in functions, we can remove this code in LLVM 18.
If rustc still needs to pass it, I'll add `PreservedSymbols` to the header file for rustc to use.

Reply:
(For future reference: #114209 *might* end up being of interest for accomplishing the desired outcome here, of keeping two otherwise disconnected lists in sync with each other in our code base.)